app/src/main/java/com/example/mylittlepokemons/MainActivity.kt

```diff
@@ -1,20 +1,44 @@
 package com.example.mylittlepokemons
 
+import android.content.Intent
 import android.os.Bundle
-import androidx.activity.enableEdgeToEdge
 import androidx.appcompat.app.AppCompatActivity
-import androidx.core.view.ViewCompat
-import androidx.core.view.WindowInsetsCompat
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.mylittlepokemons.data.Pokemon
+import com.example.mylittlepokemons.data.listOfPokemons
+import com.example.mylittlepokemons.databinding.ActivityMainBinding
 
 class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        enableEdgeToEdge()
-        setContentView(R.layout.activity_main)
-        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
-            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
-            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
-            insets
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        val pokemonAdapter = PokemonAdapter { position ->
+            val intent = Intent(this, DetailsActivity::class.java)
+            val bundle = Bundle()
+            bundle.putInt(Pokemon.POKEMON_ID_KEY, position)
+            intent.putExtras(bundle)
+            startActivity(intent)
+        }
```

Comment:
```suggestion
        val pokemonAdapter = PokemonAdapter { id ->
            val intent = Intent(this, DetailsActivity::class.java)
            intent.putIntExtra(Pokemon.POKEMON_ID_KEY, id)
            startActivity(intent)
        }
```
Или же, если хочешь бандл, то есть `bundleOf(Pokemon.POKEMON_ID_KEY to id)`